### PR TITLE
Add shortcut handling to `OptionButton`

### LINKED ISSUE #80201: OptionButton Item Shortcut is broken
### Godot version

4.1.1-stable

### System information

Windows 10

### Issue description

OptionButton items shortcut is not working

### PR DESCRIPTION
**Pull request by WhalesState**
Fixes https://github.com/godotengine/godot/issues/80201

### CODE REVIEW

File: scene/gui/option_button.cpp [hunk 1]
```diff
@@ -573,6 +589,7 @@ void OptionButton::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("is_fit_to_longest_item"), &OptionButton::is_fit_to_longest_item);
 	ClassDB::bind_method(D_METHOD("set_allow_reselect", "allow"), &OptionButton::set_allow_reselect);
 	ClassDB::bind_method(D_METHOD("get_allow_reselect"), &OptionButton::get_allow_reselect);
+	ClassDB::bind_method(D_METHOD("set_disable_shortcuts", "disabled"), &OptionButton::set_disable_shortcuts);
```
Comment (dalexeev) [inline scene/gui/option_button.cpp:592]: Bound methods must be documented. Use `bin/<godot executable> --doctool` to regenerate XML docs.

Comment (dalexeev) [reply scene/gui/option_button.cpp:592]: See the docs of `MenuBar` and `MenuButton`.

https://github.com/godotengine/godot/blob/237bd0a615df8a0e57bc3d299894abece7b43a0c/doc/classes/MenuBar.xml#L59-L65

https://github.com/godotengine/godot/blob/237bd0a615df8a0e57bc3d299894abece7b43a0c/doc/classes/MenuButton.xml#L20-L26

File: scene/gui/option_button.cpp [hunk 2]
```diff
@@ -583,9 +600,15 @@ void OptionButton::_bind_methods() {
...
 }
 
+void OptionButton::set_disable_shortcuts(bool p_disabled) {
+	disable_shortcuts = p_disabled;
+}
+
 OptionButton::OptionButton(const String &p_text) :
 		Button(p_text) {
 	set_toggle_mode(true);
+	set_disable_shortcuts(false);
```
Comment (dalexeev) [inline scene/gui/option_button.cpp:610]: This line is optional, `disable_shortcuts` is `false` by default. But I see that you copied this from `MenuButton` which has the same flaw. It's harmless, though.

**Comment by WhalesState** [comment]
maybe adding a new menu menu item and assigning it's shortcut or creating a new shortcut menu item will not work by default if i removed this line and will require to set shortcuts enabled with an extra line of code , should i really remove it ?

**Comment by AThousandShips** [comment]
Added a more descriptive title to the PR, would be good to have the commit message have an equally descriptive one, remember that it's better to describe what is done rather than what problem is fixed

**Comment by WhalesState** [comment]
this is the main reason why i have added this fix

``` c++

void OptionButton::shortcut_input(const Ref<InputEvent> &p_event) {
	ERR_FAIL_COND(p_event.is_null());

	if (disable_shortcuts) {
		return;
	}

	if (p_event->is_pressed() && !p_event->is_echo() && !is_disabled() && is_visible_in_tree() && popup->activate_item_by_event(p_event, false)) {
		accept_event();
		return;
	}

	Button::shortcut_input(p_event);
}

```

**Comment by AThousandShips** [comment]
Then I'd suggest:
```
Add shortcut handling to `OptionButton`

Also adds option to disable shortcuts
```
As the commit message

**Comment by WhalesState** [comment]
@AThousandShips choose whatever you like, just describe it to be addressed correctly in the list of fixes in godot 4.2

**Comment by AThousandShips** [comment]
No, but I don't think the current message is very helpful, as it doesn't specify what has been fixed, i.e. that no handling exists period

**Comment by WhalesState** [comment]
ok i will change it and remove the extra line that enables the enabled variable again, we also may need too change it later in MenuButton too, just remember to mention it when someone changes their files in a pull request

**Comment by AThousandShips** [comment]
You can wait until CI has finished to reduce time in next run, as it caches some of the results

**Comment by YuriSizov** [comment]
On the surface it looks correct as it mimics similar logic from `MenuButton`. Still, could you provide a test project that you've used to validate the change?

**Comment by WhalesState** [comment]
i have noticed the issue while working on Editor in c++, for example CanvasItemEditor is using MenuButtons with shortcuts for snapping options, but when you try to use the OptionButton with shortcuts it doesn't work because it was missing shortcut handeling function.

this single line will work fine with MenuButton's PopupMenu while it won't work with OptionButton's PopupMenu

``` c++

select_popup->set_item_shortcut(TOOL_SELECT, ED_SHORTCUT("canvas_item_editor/select_mode", TTR("Select Mode"), Key::Q));

```

**Comment by WhalesState** [comment]
This is how i tested it, which will be hard to be provided as test project at the moment since it's for [godot-2d](https://github.com/WhalesState/godot-2D/tree/GLES3) repo
image
also ignore that the shortcuts offset in tooltip are drawn with wrong offset, this is an editor theme issue and will try to fix it too.

**Comment by WhalesState** [comment]
have found this which may be helpful for testing, just create an option button and add this script to it then press Q and it will change the item

```gdscript

extends OptionButton


func _ready():
	add_item("test")
	add_item("test with shortcut")
	var sc := Shortcut.new()
	var ev := InputEventKey.new()
	ev.keycode = KEY_Q
	sc.events.append(ev)
	get_popup().set_item_shortcut(1, sc, true)


```

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**